b2sdk/api.py

```diff
@@ -312,18 +324,21 @@ def list_buckets(self, bucket_name=None, bucket_id=None):
         """
         # Give a useful warning if the current application key does not
         # allow access to the named bucket.
-        self.check_bucket_restrictions(bucket_name)
+        assert not (bucket_name and bucket_id)
```

Comment:
this is parameter constraint *validation* - we should not validate using asserts and AssertionError is wrong error to raise here - ValueError would be better